<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Future Innovators Mentorship</title>
    <meta charset="utf-8" />
    <meta name="author" content="João Vitor Cavalcante" />
    <script src="Future-Innovators-Mentorship_files/header-attrs/header-attrs.js"></script>
    <link href="Future-Innovators-Mentorship_files/animate.css-xaringan/animate.fade.css" rel="stylesheet" />
    <link rel="stylesheet" href="custom.css" type="text/css" />
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Future Innovators Mentorship
]
.subtitle[
## BioData Developers Network
]
.author[
### João Vitor Cavalcante
]
.date[
### May 16th, 2024
]

---








class: inverse center middle

# Software Project

---

# beiko-lab/ARETE

- A bioinformatics best-practice [analysis pipeline](https://github.com/beiko-lab/arete/) for profiling the genomic repertoire and evolutionary dynamics of microorganisms with a particular focus on pathogens.
  - Pangenome inference, coevolution and lateral gene transfer analysis are some of its features. 

- Originally ideated by Finlay Maguire and Alex Manuele in 2021, mostly implemented by me from 2023 onwards.

- Intended to run in HPC environments, through Docker or Singularity.

&lt;center&gt;&lt;img src="https://raw.githubusercontent.com/beiko-lab/arete/master/assets/arete-logo-transparent.png" alt="ARETE logo" width="600px" /&gt;&lt;/center&gt;
---

![ARETE diagram](diagram.png)

---

# Some highlights

- Based on a customized version of the [nf-core Pipeline template](https://github.com/nf-core/tools?tab=readme-ov-file#creating-a-new-pipeline), same as the Tree of Life workflows.

- Public [documentation website](https://beiko-lab.github.io/arete/) built with [MkDocs](https://www.mkdocs.org/) and served through GitHub Pages.

- Full test suite for every subworkflow included in the pipeline, implemented in [nf-test](https://www.nf-test.com/) and executed through GitHub Actions.
  - Standard end-to-end testing is also executed in GitHub actions, for both Singularity and Docker.
  
- Different [resource allocation profiles](https://github.com/beiko-lab/arete/tree/master/conf), allowing for fine-grained customization in different environments.

- Contributions to the nf-core community, seen through modules such as [IslandPath](https://nf-co.re/modules/islandpath) and [PhiSpy](https://nf-co.re/modules/phispy), both originally implemented in this pipeline.

&lt;center&gt;
&lt;img src="https://avatars.githubusercontent.com/u/6698688?s=280&amp;v=4" alt="nextflow logo" width="100px" /&gt;
&lt;img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fopensource.scilifelab.se%2Fwp-content%2Fuploads%2F2018%2F08%2Fnf-core-logo-square.png&amp;f=1&amp;nofb=1&amp;ipt=1f50d06bb2787ce1dbafdedd7626086bcc175f11a7cba59550a2b295d7931f24&amp;ipo=images" alt="nf-core logo" width="100px" /&gt;
&lt;img src="https://seeklogo.com/images/G/github-actions-logo-031704BDC6-seeklogo.com.png" alt="GithHub Actions logo" width="100px" /&gt;
&lt;/center&gt;

---
class: inverse center middle

# Mentorship Project
## Genome After-Party:
### A global database of ready-made genome analyses

---

## What skills can I contribute to the project?

.font-larger[
- I have deep knowledge of Nextflow and nf-core;

- I have extensively worked in the past, through both ARETE and other projects, like [EURYALE](https://github.com/dalmolingroup/euryale), in implementing analyses from researchers in workflows;

- I have implemented many wrappers in both Python and Shell in workflows I've developed, like ARETE itself;

- Through both my MSc and Undergrad research, I have become acquainted with many bioinformatics tools, which will facilitate getting started integrating them in curated workflows.
]

---
class: middle center

# Future Innovators Mentorship

## BioData Developers Network

### João Vitor Cavalcante

### May 16th, 2024
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"slideNumberFormat": "%current%/%total%",
"highlightStyle": "solarized-dark",
"highlightLines": true,
"ratio": "16:9",
"countIncrementalSlides": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
