---
title: "Single-cell RNA-Seq do Córtex Visual de Camundongos"
author: "João Vitor Ferreira"
institute: "Dalmolin Lab, BioME"
date: "28-06-2021"
output:
  xaringan::moon_reader:
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
    css: xaringan-themer.css
    seal: false
---
class: center

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
```

```{r, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, fig.retina = 3, out.width = "80%", out.height="450")
library(DT)
library(dplyr)
library(ggplot2)
library(readr)
library(purrr)
enriched_kegg_ne_vs_1h <- read_tsv("/home/jvfe/dev/Pesquisa/neuron-visual-cortex/results/enrichment_results/kegg_enriched_unst_vs_1hour.tsv")
enriched_kegg_ne_vs_4h <- read_tsv("/home/jvfe/dev/Pesquisa/neuron-visual-cortex/results/enrichment_results/kegg_enriched_unst_vs_4hour.tsv")
enriched_kegg_h1_vs_h4 <- read_tsv("/home/jvfe/dev/Pesquisa/neuron-visual-cortex/results/enrichment_results/kegg_enriched_onehour_vs_4hour.tsv")


enriched_go_ne_vs_1h <- read_tsv("/home/jvfe/dev/Pesquisa/neuron-visual-cortex/results/enrichment_results/go_enriched_unst_vs_1hour.tsv")
enriched_go_ne_vs_4h <- read_tsv("/home/jvfe/dev/Pesquisa/neuron-visual-cortex/results/enrichment_results/go_enriched_unst_vs_4hour.tsv")
enriched_go_h1_vs_h4 <- read_tsv("/home/jvfe/dev/Pesquisa/neuron-visual-cortex/results/enrichment_results/go_enriched_onehour_vs_4hour.tsv")
```


```{r funcs, include=FALSE}
get_n_transcripts <- function(diffexp_table) {
  diffexp_table %>%
    pull(row) %>%
    unique() %>%
    length()
}
aggregate_transcripts <- function(data) {
  data %>%
    select(ensembl_transcript_id_version,
           ensembl_gene_id,
           mgi_symbol,
           description) %>%
    group_by(ensembl_gene_id, mgi_symbol, description) %>%
    summarise(transcript_ids = paste(ensembl_transcript_id_version, collapse = "; ")) %>% 
    setNames(c("Ensembl Gene ID", "MGI Symbol", "Gene Description", "Trancritos Dif.Exp."))
}
calculate_gene_ratio <- function(data) {
  data %>%
    tidyr::separate(GeneRatio, c("n_genes", "n_total")) %>%
    mutate(gene_ratio = Count / as.numeric(n_total))
}
plot_barplot <- function(data) {
  data %>%
    calculate_gene_ratio(.) %>%
    slice_max(gene_ratio, n = 15, with_ties = FALSE) %>%
    arrange(gene_ratio) %>%
    mutate(Description = factor(Description, Description)) %>%
    ggplot(aes(
      x = Description,
      y = gene_ratio,
      color = pvalue,
      size = Count
    )) +
    geom_point() +
    scale_color_continuous(high = "blue", low = "red") +
    coord_flip() +
    theme_light(base_family = "Roboto") +
    labs(x = NULL, y = NULL)
}
make_datatable <- function(data, col_names) {
  data %>%
    datatable(
      .,
      rownames = FALSE,
      escape = FALSE,
      colnames = col_names,
      filter = 'top',
      options = list(
        lengthMenu = c(5, 10, 25, 50, 100),
        pageLength = 10,
        scrollY = 600,
        scroller = TRUE,
      )
    )
}
```

```{r xaringan-themer, include=FALSE, warning=FALSE}
library(xaringanthemer)
style_duo(
  primary_color = "#F97B64", secondary_color = "#1F4257",
  header_font_google = google_font("Josefin Sans"),
  text_font_google   = google_font("Montserrat", "300", "300i"),
  code_font_google   = google_font("Fira Mono")
)

result_files <-
  grep(
    "translated",
    list.files("/home/jvfe/dev/Pesquisa/neuron-visual-cortex/results/deseq_results", pattern = ".tsv", full.names = TRUE),
    invert = TRUE,
    value = TRUE
  )
file_names <- stringr::str_remove(result_files, pattern = "^(.*[\\/])")
diff_exp <- map(result_files, read_tsv)
names(diff_exp) <- file_names
summarized <- map(diff_exp, ~ get_n_transcripts(.x)) %>% 
  bind_rows() %>%
  tidyr::pivot_longer(
    everything()
  ) %>%
  mutate(Contraste = c("1h vs 4h", "N.E. vs 1h", "N.E. vs 4h")) %>%
  arrange(desc(value)) %>%
  select(-c(name)) %>%
  select(Contraste, value)
```

# Single-cell RNA-Seq do Córtex Visual de Camundongos

### João Vitor Ferreira
### Dalmolin Lab, BioME + UFRGS
**28-06-2021**

---
class: inverse

## Proposta Inicial

* Determinar diferenças transcricionais entre um neurônio ativo e inativo

* Nenhum dataset específico encontrado!

* Porém, GSE102827 pareceu promissor.

---
## O estudo

* Sequencia córtex visual de *Mus musculus* usando o protocolo inDrops;

* Três contrastes: Não estimulado visualmente (controle), 
estimulado com estímulo luminoso por 1 hora e
estimulado com estímulo luminoso por 4 horas.

* Focou sobretudo na identificação de marcadores neuronais para estímulo luminoso e na comparação entre diferentes subtipos celulares,
porém também realizaram enriquecimento GO e GSEA.

---
class: inverse

## Os Resultados

```{r}
summarized %>% 
  knitr::kable(
    format = "html",
    align = "cc",
    col.names = c("Contraste", "Número de transcritos Dif.Exp."),
    table.attr = 'class = "table"'
  )
```

---
class: center

## Enriquecimentos - KEGG

```{r xaringan-panelset, echo=FALSE}
xaringanExtra::use_panelset()
```

.panelset[
.panel[.panel-name[KEGG-N.E. vs 1h]
Nenhum resultado significativo
]

.panel[.panel-name[KEGG-N.E. vs 4h]

```{r panel-chunk1}
plot_barplot(enriched_kegg_ne_vs_4h)
```
]

.panel[.panel-name[KEGG-1h vs 4h]

```{r panel-chunk2}
plot_barplot(enriched_kegg_h1_vs_h4)
```
]
]

---
class: center, inverse

## Enriquecimentos - GO

```{r xaringan-panelset2, echo=FALSE}
xaringanExtra::use_panelset()
```

.panelset[
.panel[.panel-name[GO-N.E. vs 1h]
```{r panel-chunk3}
plot_barplot(enriched_go_ne_vs_1h)
```
]

.panel[.panel-name[GO-N.E. vs 4h]

```{r panel-chunk5}
plot_barplot(enriched_go_ne_vs_4h)
```
]

.panel[.panel-name[GO-1h vs 4h]

```{r panel-chunk6}
plot_barplot(enriched_go_h1_vs_h4)
```
]
]

---
class: center

# Single-cell RNA-Seq do Córtex Visual de Camundongos

### João Vitor Ferreira
### Dalmolin Lab, BioME + UFRGS
**28-06-2021**